src/main/java/net/sf/jabref/logic/cleanup/BiblatexCleanup.java

```diff
@@ -35,22 +35,26 @@
         }
 
         // Dates: create date out of year and month, save it and delete old fields
-        entry.getField(FieldName.DATE).ifPresent(date -> {
-            if (date.isEmpty()) {
-                entry.getFieldOrAlias(FieldName.DATE).ifPresent(newDate -> {
-                    Optional<String> oldYear = entry.getField(FieldName.YEAR);
-                    Optional<String> oldMonth = entry.getField(FieldName.MONTH);
-
-                    entry.setField(FieldName.DATE, newDate);
-                    entry.clearField(FieldName.YEAR);
-                    entry.clearField(FieldName.MONTH);
-
-                    changes.add(new FieldChange(entry, FieldName.DATE, null, newDate));
-                    changes.add(new FieldChange(entry, FieldName.YEAR, oldYear.orElse(null), null));
-                    changes.add(new FieldChange(entry, FieldName.MONTH, oldMonth.orElse(null), null));
-                });
-            }
-        });
+        // If there already exists a value for the field date, it is not overwritten
+        String date = "";
+        if (entry.getField(FieldName.DATE).isPresent()) {
+            date = entry.getField(FieldName.DATE).get();
+        }
+        if (date.isEmpty()) {
```

Comment:
I used this approach, because it captures 2 purposes and checks the optonal
a) If the field is not present the date is empty
b) if the field is present, we assign the value, however, it still might be empty